mlflow/models/python_api.py

```diff
@@ -0,0 +1,200 @@
+           - single input payload, when content type is "json".
+           - Pandas DataFrame, when content type is "csv".
+        - A Python list. The content type has to be "csv".
+        - A Pandas DataFrame. The content type has to be "csv".
+        - A string represents serialized input data. e.g. '{"inputs": [1, 2]}'
+        - None to input data from stdin.
+
+    :param input_path: Path to a file containing input data. If provided, 'input_data' must be None.
+    :param content_type: Content type of the input data. Can be one of {‘json’, ‘csv’}.
+    :param output_path: File to output results to as json. If not provided, output to stdout.
+    :param env_manager: Specify a way to create an environment for MLmodel inference:
+
+        - virtualenv (default): use virtualenv (and pyenv for Python version management)
+        - local: use the local environment
+        - conda: use conda
+
+    :param install_mlflow: If specified and there is a conda or virtualenv environment to be
+        activated mlflow will be installed into the environment after it has been activated.
+        The version of installed mlflow will be the same as the one used to invoke this command.
+    :param pip_requirements_override: If specified, install the specified python dependencies to
+        the model inference environment. This is particularly useful when you want to add extra
+        dependencies or try different versions of the dependencies defined in the logged model.
+
+    Code example:
+
+    .. code-block:: python
+
+        import mlflow
+
+        run_id = "..."
+
+        mlflow.pyfunc.predict(
+            model_uri=f"runs:/{run_id}/model",
+            input_data={"x": 1, "y": 2},
+            content_type="json",
+        )
+
+        # Run prediction with addtioanl pip dependencies
+        mlflow.pyfunc.predict(
+            model_uri=f"runs:/{run_id}/model",
+            input_data='{"x": 1, "y": 2}',
+            content_type="json",
+            pip_requirements_override=["scikit-learn==0.23.2"],
+        )
+
+    """
+    if content_type not in [_CONTENT_TYPE_JSON, _CONTENT_TYPE_CSV]:
+        raise MlflowException.invalid_parameter_value(
+            f"Content type must be one of {_CONTENT_TYPE_JSON} or {_CONTENT_TYPE_CSV}."
+        )
+
+    def _predict(_input_path: str):
+        return get_flavor_backend(
+            model_uri, env_manager=env_manager, install_mlflow=install_mlflow
+        ).predict(
+            model_uri=model_uri,
+            input_path=_input_path,
+            output_path=output_path,
+            content_type=content_type,
+            pip_requirements_override=pip_requirements_override,
+        )
+
+    if input_data is not None and input_path is not None:
+        raise MlflowException.invalid_parameter_value(
+            "Both input_data and input_path are provided. Only one of them should be specified."
+        )
+    elif input_data is not None:
```

Comment:
Not `input_data` as this can be pandas dataframe, which doesn't allow casting to single boolean.